### PR TITLE
Final preparations for 3.5.0 release

### PR DESCRIPTION
**Pull request by mattpap**
- [x] updated `switcher.json`
- [x] updated release notes

### CODE REVIEW

**Review by bryevdv — APPROVED** [review]

**Comment by mattpap** [comment]
Just to note, WIP status is because I'm going through the list of PRs for 3.5 to make sure everything relevant was included. Given the approval, I will merge when I'm finished with this, and then will restart the release build.

File: docs/bokeh/switcher.json [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
   {
     "name": "latest",
-    "version": "3.4.1",
+    "version": "3.5.0",
     "url": "https://docs.bokeh.org/en/latest/",
     "preferred": true
   },
+  {
+    "version": "3.4.1",
+    "url": "https://docs.bokeh.org/en/3.4.1/"
```
Comment (mosc9575) [inline docs/bokeh/switcher.json]:
```suggestion
    "version": "3.4.2",
    "url": "https://docs.bokeh.org/en/3.4.2/"
```

Not sure what happens to the `3.4.3` point release anyhow the latest point release is not `3.4.1`.

Comment (bryevdv) [reply docs/bokeh/switcher.json]: thanks for the catch @mosc9575

File: docs/bokeh/switcher.json [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+  },
   {
     "version": "3.3.0",
     "url": "https://docs.bokeh.org/en/3.3.0/"
```
Comment (mosc9575) [inline docs/bokeh/switcher.json]:
```suggestion
    "version": "3.3.4",
    "url": "https://docs.bokeh.org/en/3.3.4/"
```

This was mentioned in https://github.com/bokeh/bokeh/pull/13929#discussion_r1638361397 the first time. The latest point release for the `3.3` series is `3.3.4`.

File: docs/bokeh/source/docs/releases/3.5.0.rst
Comment (mosc9575) [inline docs/bokeh/source/docs/releases/3.5.0.rst:1]: @bryevdv mentioned in https://github.com/bokeh/bokeh/pull/13929#issuecomment-2178827503.

> all `3.4.x` point release notes are missing from branch-3.5.

This isn't fixed yet.

**Pull request merged**